testdata/dnn/onnx/generate_onnx_models.py

```diff
@@ -1356,3 +1356,47 @@ def forward(self, x):
 x = Variable(torch.randn([2, 3]))
 model = NormalizeFusion()
 save_data_and_model("normalize_fusion", x, model)
+
+class CumSum(nn.Module):
+    def __init__(self, dim):
+        super(CumSum, self).__init__()
+        self._dim = dim
+
+    def forward(self, x):
+        return torch.cumsum(x, self._dim)
+
+x = torch.randn(2, 3)
+save_data_and_model("cumsum_2d_dim_1", x, CumSum(dim=1), version=11)
+
+x = torch.randn(2, 3, 4)
+save_data_and_model("cumsum_3d_dim_2", x, CumSum(dim=2), version=11)
+
+# Torch does not support cumsum "exclusive" and "reverse" attributes so they are manually added.
```

Comment:
Can you try to convert this network from another framework to onnx instead of manually building the graph? keras2onnx 1.7 + TF 2.5 seems to work fine with [this](https://github.com/onnx/keras-onnx/pull/700/files) patch.

Reply:
Thanks for the review.  I regenerated the models from tf via tf2onnx, which does support exclusive and reverse flags.

I didn't add the code generation as `generate_onnx_models.py` doesn't have tf and tf2onnx setup.  Let me know if you want to include the code generation in some other file or a new file.

Reply:
Thanks for the review. I regenerated the models from tf via tf2onnx, which does support exclusive and reverse flags.

I didn't add the code generation as generate_onnx_models.py doesn't have tf and tf2onnx setup. Let me know if you want to include the code generation in some other file or a new file.

```
>>> import onnx
>>> model = onnx.load("models/cumsum_1d_reverse.onnx")
>>> model
ir_version: 7
producer_name: "tf2onnx"
producer_version: "1.9.1"
graph {
  node {
    input: "x"
    input: "PartitionedCall/Cumsum/axis:0"
    output: "output_0"
    name: "PartitionedCall/Cumsum"
    op_type: "CumSum"
    attribute {
      name: "exclusive"
      i: 0
      type: INT
    }
    attribute {
      name: "reverse"
      i: 1
      type: INT
    }
  }
  name: "tf2onnx"
  initializer {
    data_type: 6
    name: "PartitionedCall/Cumsum/axis:0"
    raw_data: "\000\000\000\000"
  }
  doc_string: "converted from cumsum_1d_reverse/"
  input {
    name: "x"
    type {
      tensor_type {
        elem_type: 1
        shape {
        }
      }
    }
  }
  output {
    name: "output_0"
    type {
      tensor_type {
        elem_type: 1
        shape {
        }
      }
    }
  }
}
opset_import {
  version: 14
}

```

Reply:
Thanks for the review. I regenerated the models from tf via tf2onnx, which does support exclusive and reverse flags.  An example is below:

I didn't add the code generation as generate_onnx_models.py doesn't have tf and tf2onnx setup. Let me know if you want to include the code generation in some other file or a new file.

```
>>> import onnx
>>> model = onnx.load("models/cumsum_1d_reverse.onnx")
>>> model
ir_version: 7
producer_name: "tf2onnx"
producer_version: "1.9.1"
graph {
  node {
    input: "x"
    input: "PartitionedCall/Cumsum/axis:0"
    output: "output_0"
    name: "PartitionedCall/Cumsum"
    op_type: "CumSum"
    attribute {
      name: "exclusive"
      i: 0
      type: INT
    }
    attribute {
      name: "reverse"
      i: 1
      type: INT
    }
  }
  name: "tf2onnx"
  initializer {
    data_type: 6
    name: "PartitionedCall/Cumsum/axis:0"
    raw_data: "\000\000\000\000"
  }
  doc_string: "converted from cumsum_1d_reverse/"
  input {
    name: "x"
    type {
      tensor_type {
        elem_type: 1
        shape {
        }
      }
    }
  }
  output {
    name: "output_0"
    type {
      tensor_type {
        elem_type: 1
        shape {
        }
      }
    }
  }
}
opset_import {
  version: 14
}

```

Reply:
Feel free to add generating code in generate_onnx_models.py and leave a comment with tf/tf2onnx versions.